src/Serilog/Configuration/LoggerSettingsConfiguration.cs

```diff
@@ -52,7 +54,9 @@ public LoggerConfiguration Settings(ILoggerSettings settings)
         public LoggerConfiguration KeyValuePairs(IEnumerable<KeyValuePair<string, string>> settings)
         {
             if (settings == null) throw new ArgumentNullException(nameof(settings));
-            return Settings(new KeyValuePairSettings(settings));
+
+            var settingsDictionary = new ReadOnlyDictionary<string, string>(settings.ToDictionary(x => x.Key, x => x.Value));
+            return Settings(new KeyValuePairSettings(settingsDictionary));
```

Comment:
I think `Dictionary` implements `IReadOnlyDictionary` in recent framework versions - I don't think the creation of the `ReadOnlyDictionary` should be needed here.

Reply:
Ah yes, that's true. It's one of those strange things where a *writable* dictionary actually implements *read-only* dictionary .... 

I'm not sure what the "best practices" are in that regard, but I guess in internal classes there is no need to be a control-freak :) 

I'll change that.